File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
     HashMap<Integer,Integer> gg=new HashMap<>();
     ArrayList<Integer> hh[]=new ArrayList[nums.length];
     for(int i:nums)
     {
        if(gg.containsKey(i))
        {
            gg.put(i,gg.get(i)+1);
        }
        else
        {
            gg.put(i,1);
        }
        
     } 
     for(int i:gg.keySet())
     {  if(hh[gg.get(i)-1]==null){
        hh[gg.get(i)-1]=new ArrayList<>();
     }
        hh[gg.get(i)-1].add(i);
        
     }
     ArrayList<Integer> jj=new ArrayList<>();
     for(int i=nums.length-1;i>-1;i--)
     {  if(hh[i]!=null)
     {
        jj.addAll(hh[i]);
     }
     
        
     }
     int[] arr=new int[k]; 
     for(int i=0;i<k;i++)
     {
        arr[i]=jj.get(i);
     }
     
     
   return arr; }
}

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> map;
        priority_queue<pair<int,int>,vector<pair<int,int>>> pq;
        vector<int> answer;

        for(int i:nums){map[i]++;}

        for(auto& both:map){
            pq.push({both.second,both.first});
        }

        for(int i=0;i<k;i++){
            if(pq.empty()){break;}
            answer.push_back(pq.top().second);
            pq.pop();
        }

        return answer;
    }
};